### PR TITLE
Docs: specify that language-server table must be before language and grammar array

### PR DESCRIPTION
**Pull request by gabydd**
Due to a limitation in toml, when specifying the new config  for language servers from #2507 because they are just tables they must be put before the arrays of tables (language and grammar). I have seen a couple issues and matrix questions pop up where people didn't know this and put the language server configuration after the language config and then nothing happens, #7149 and https://github.com/helix-editor/helix/discussions/7101#discussioncomment-5962068. If we think there is going to be more arrays of tables in the future we could make the message more general but I think this way is more clear. I couldn't find any official documentation about this behaviour of toml but I am pretty sure it is the case.

### CODE REVIEW

**Comment by gabydd** [comment]
Similarly people have started updating https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers with the new language server config but specificying it below the language table which should probably be fixed as well. It's late here so I will do that in the morning though.

File: book/src/languages.md [hunk 1]
```diff
@@ -99,6 +99,9 @@ with the following priorities:
 
 Language servers are configured separately in the table `language-server` in the same file as the languages `languages.toml`
 
+> 💡 Due to limitations of toml anything in the ``language-server`` table must
+> be specified before the ``[[language]]`` and ``[[grammar]]`` arrays of tables.
```
Comment (the-mikedavis) [inline book/src/languages.md:103]:
```suggestion
> 💡 Due to limitations of TOML, anything in the `language-server` table must
> be specified before the `[[language]]` and `[[grammar]]` arrays of tables.
```

I think the double backticks are not necessary in markdown even in blockquotes

**Comment by gabydd** [comment]
this might not be the best place to ask this but I wonder if we should put a note on the installing language servers wiki page that we are moving to a new configuration format for language servers and then start changing the wiki to the new format. Or wait till the next release to move the wiki over to the new format. We could still have a note but just saying something like: "The configuration on this page is only for the current release here is how to use it with the current master."

**Comment by gabydd** [comment]
been looking at this again and it seems like you don't actually need to put the language-server table above any arrays of tables, I am guessing something else was happening in those issues and I incorrectly assumed that the moving the ``[[language]]`` table fixed it. I am going to close this pr unless someone can reproduce this problem.

**Comment by gabydd** [comment]
ah this was a problem when adding the new language server when running. then reloading config, and then restarting the server. If it seems possible to fix I'll make an issue for it.

**Pull request closed without merging**